tools/wpt/run.py

```diff
@@ -487,6 +492,23 @@ def setup_wptrunner(venv, prompt=True, install_browser=False, **kwargs):
     setup_cls = product_setup[kwargs["product"]](venv, prompt, sub_product)
     setup_cls.install_requirements()
 
+    affected_revish = kwargs.pop("affected", None)
+    if affected_revish is not None:
+        files_changed, _ = testfiles.files_changed(
+            affected_revish, include_uncommitted=True, include_new=True)
+        # TODO(Hexcles): Perhaps use wptrunner.testloader.ManifestLoader here
+        # so that we can reuse it later in wptrunner instead of loading it
+        # again and remove the manifest-related code from testfiles.
+        tests_changed, tests_affected = testfiles.affected_testfiles(
+            files_changed, manifest_path=kwargs.get("manifest_path"), manifest_update=kwargs["manifest_update"])
+        test_list = tests_changed | tests_affected
+        logger.info("Identified %s affected tests" % len(test_list))
+        if not test_list:
+            logger.info("Quitting because no tests were affected.")
```

Comment:
This is a great question that I didn't realize. After some experiments, it seems the file `wptreport.json` will always be created (right after logging is set up), but it will be an empty file if no tests run. Hence, the Taskcluster job will not fail, and we will get a weird `wptreport.json.gz` that's empty. An empty file cannot be succesfully deserialized by Python `json`, so this will eventually become an invalid report and will be discarded by the results processor.

@jgraham is this behaviour (producing an empty, invalid JSON report) acceptable? Or do you have other suggestions?

Reply:
For context, gh-9481 tracks that bug, and in the mean time [we've implemented a workaround for results-collection project](https://github.com/web-platform-tests/results-collection/blob/1b5dcf090e0bd6a2443ed6f1cb253e548c8de87c/src/scripts/run-and-verify.py#L116-L128).

Reply:
It seems like we shouldn't let that block this issue at least.